manager/set_manager.py

```diff
@@ -0,0 +1,69 @@
+from .fridge_manager import FridgeManager
+from decorators.decorators import count_args
+from decorators.decorators import calls_limit
+
+
+class SetManager:
+    def __init__(self, manager: FridgeManager):
+        """
+        Constructor for SetManager objects.
+
+        Args:
+            manager (FridgeManager): The FridgeManager object to be associated with SetManager.
+        """
+        self.manager = manager
+        self.index = 0
+
+    @calls_limit
+    def __len__(self):
+        """
+        Calculates and returns the total number of color_available across all fridges in the associated FridgeManager.
+
+        Returns:
+            int: The total number of color_available.
+        """
+        length = 0
+        for fridge in self.manager:
+            length += len(fridge.color_available)
+        return length
+
+    @calls_limit
+    def __iter__(self):
+        """
+        Iterates over the color_available across all fridges in the associated FridgeManager.
+
+        Yields:
+            Any: The next available color.
+        """
+        for fridge in self.manager:
+            for color_available in fridge.color_available:
+                yield color_available
```

Comment:
the problem is that there would be duplicates. For example, if you have two kitchens with same type, their chefs would be same